### PR TITLE
Handle clangWarnings as errors, handle script and linker errors

### PR DESCRIPTION
**Pull request by ecamacho**
I added more error type's categorisation. I also moved the parsing logic out to its own extension.

cc @alrocha @BalestraPatrick @polac24

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -66,6 +66,11 @@ task :gen_resources do
   gen_fake_resources
 end
 
+task :test_version do
+  version = get_version
+  p version
```
Comment (BalestraPatrick) [inline Rakefile:71]: Is this still useful?

Comment (ecamacho) [reply Rakefile]: Not at all, I removed it

File: Sources/XCLogParser/parser/Notice+Parser.swift [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    /// - returns: An Array of `Notice`
+    public static func parseFromLogSection(_ logSection: IDEActivityLogSection, forType type: DetailStepType)
+        -> [Notice] {
+        // we look for clangWarnings parsing the text of the logSection
+        var clangWarnings: [Notice] = []
+        if let clangWarningsFlags = self.parseClangWarningFlags(text: logSection.text),
+            clangWarningsFlags.count > 0 {
+            clangWarnings = zip(logSection.messages, clangWarningsFlags)
+                .compactMap { (message, warningFlag) -> Notice? in
+                    // If the warning is treated as error, we marked the issue as errro
```
Comment (polac24) [inline Sources/XCLogParser/parser/Notice+Parser.swift:39]:
```suggestion
                    // If the warning is treated as error, we marked the issue as error
```

File: Sources/XCLogParser/parser/Notice+Parser.swift [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                }
+                // Ld command errors
+                if notice.type == .error && type == .linker {
+                    return [notice.with(type: .linkerError)]
+                }
+                // Build phase's script errors
+                if notice.type == .scriptPhaseError {
+                    // Decorate script phase error with the signature that contains the name of the
+                    // phase and the target
+                    return [notice.with(detail: "\(notice.detail ?? "") \(logSection.signature)")]
```
Comment (polac24) [inline Sources/XCLogParser/parser/Notice+Parser.swift:89]: 👍

**Pull request merged**